src/Doctrine/Annotation/Tokens.php

```diff
@@ -240,23 +240,21 @@ public function insertAt(int $index, Token $token): void
         $this[$index] = $token;
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @throws \InvalidArgumentException
-     */
     public function offsetSet($index, $token): void
     {
+        // @phpstan-ignore-next-line as we type checking here
```

Comment:
```suggestion
        /** @var mixed $token */
```
Also, there's `treatPhpDocTypesAsCertain` option in PHPStan configuration, which would prevent it from reporting runtime type checks for variables that are typed in phpdocs, although given that the entire project is statically analyzed, I think turning it off would do more harm than good.

Reply:
Switching it off on raises a 3 items on this PR, maybe it is worth doing that? Not sure....

Reply:
I don't think so. In most methods we trust PHPDoc types. If there are only a few places where we want to validate them at runtime, it's safer to add `@var mixed`, as that is how we treat this variable in this case. Switching `treatPhpDocTypesAsCertain` off would prevent PHPStan from reporting truly unnecessary checks that are there by mistake.